builtins: create function to aggregate aggregated stmt metadata

xinhaoz opened the pull request:
## server: prefer string concat over str format where possible

A number of places in the combined stats api were using string
format where a string concatenation would have worked.

Release note: None

Epic: none

## server: format sql activity queries

Reformat queries for consistency and readability.

Epic: none

Release note: None


## builtins: create function to aggregate aggregated stmt metadata
The statement activity table used to cache aggregated stmt stats for the sql activity page stores aggregated metadata. Previously we used the same query as the one used for system.statement_statistics, which stores unaggregated metadata, on the activity table. Using the aggregate function for unaggregated metadata on aggregated metadata was nto valid, leading to incorrect JSON fields being produced. This commit introduces the builtin `crdb_internal.combine_aggregated_stmt_metadata` which can be used to correctly aggregate the  metadata column on the stmt activity table.

Fixes: #103895

Release note (bug fix): On the UI, selecting a database filter from the filters menu in the sql activity page should function as expected. This fixes a preivous bug where the filter would break and not show any results when the results were retrieved from the stmt activity table instead of the persisted table.

xinhaoz committed:
builtins: create function to aggregate aggregated stmt metadata

The statement activity table used to cache aggregated stmt stats for the
sql activity page stores aggregated metadata. Previously we used the
same query as the one used for system.statement_statistics, which stores
unaggregated metadata, on the activity table. Using the aggregate function
for unaggregated metadata on aggregated metadata was nto valid, leading to
incorrect JSON fields being produced. This commit introduces the builtin
`crdb_internal.combine_aggregated_stmt_metadata` which can be used to
correctly aggregate the  metadata column on the stmt activity table.

Fixes: #103895

Release note (bug fix): On the UI, selecting a database filter
from the filters menu in the sql activity page should function
as expected. This fixes a preivous bug where the filter would
break and not show any results when the results were retrieved
from the stmt activity table instead of the persisted table.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/105351)
<!-- Reviewable:end -->

xinhaoz committed:
server: prefer string concat over str format where possible

A number of places in the combined stats api were using string
format where a string concatenation would have worked.

Release note: None

Epic: none

xinhaoz committed:
server: format sql activity queries

Reformat queries for consistency and readability.

Epic: none

Release note: None

maryliag reviewed (approved):
Your PR has 3 commits, so on the PR description add more information about the changes on the other 2 commits.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 5 of 5 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/105351#-:-NZX94-eCUZZhpZSsDs6:bocqeu2)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @xinhaoz)

___
*[`pkg/server/combined_statement_stats.go` line 1513 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/105351#-NZXA6HW8DTNt9J_2iG1:-NZXA6HW8DTNt9J_2iG2:b-iiseaw) ([raw file](https://github.com/cockroachdb/cockroach/blob/dad7cc3b061af66e9e13e06ae86533527dfab3c7/pkg/server/combined_statement_stats.go#L1513)):*
> ```Go
> 	// We will have 1 open iterator at a time. For the deferred close operation, we will
> 	// only close the iterator if there were no errors creating the iterator. Closing an
> 	// iterator that returned an error on creation will cause a nil pointer deref.
> ```

nit: defer


<!-- Sent from Reviewable.io -->

xinhaoz commented:
TFTR!
bors r+
